packages/@uppy/core/src/index.js

```diff
@@ -1317,11 +1309,18 @@ class Uppy {
         return this._runUpload(uploadID)
       })
       .catch((err) => {
+        const message = typeof err === 'object' ? err.message : err
+        const details = (typeof err === 'object' && err.details) ? err.details : ''
+
         if (err.isRestriction) {
           this.emit('restriction-failed', null, err)
-          return onError(err)
+          this.log(`${message} ${details}`, 'info')
+          this.info({ message: message, details: details }, 'info', 5000)
```

Comment:
Do we not need to `throw` the error in case of restriction too (so that the promise rejects)?